editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx

```diff
@@ -300,20 +301,59 @@ export const useDelayedStrategyCursor = () => {
   return useDelayedStrategy<CSSCursor | null>(selector)
 }
 
+const notResizableControls: ControlWithKey = {
+  control: NonResizableControl,
+  key: 'not-resizable-control',
+  show: 'visible-except-when-other-strategy-is-active',
+}
+
+export function getApplicableControls(
+  currentStrategy: CanvasStrategyId | null,
+  strategy: CanvasStrategy,
+): Array<ControlWithKey> {
+  return strategy.controlsToRender.filter((control) => {
+    return (
+      control.show === 'always-visible' ||
+      (control.show === 'visible-only-while-active' && strategy.id === currentStrategy) ||
+      (control.show === 'visible-except-when-other-strategy-is-active' &&
+        (currentStrategy == null || strategy.id === currentStrategy))
+    )
+  })
+}
+
+export function isResizableStrategy(canvasStrategy: CanvasStrategy): boolean {
+  switch (canvasStrategy.id) {
+    case 'ABSOLUTE_RESIZE_BOUNDING_BOX':
+    case 'KEYBOARD_ABSOLUTE_RESIZE':
+      return true
+    default:
+      return false
+  }
+}
+
 export function useGetApplicableStrategyControls(): Array<ControlWithKey> {
   const applicableStrategies = useGetApplicableStrategies()
   const currentStrategy = useDelayedCurrentStrategy()
   return React.useMemo(() => {
-    return applicableStrategies.reduce<ControlWithKey[]>((working, s) => {
-      const filteredControls = s.controlsToRender.filter(
-        (control) =>
-          control.show === 'always-visible' ||
-          (control.show === 'visible-only-while-active' && s.id === currentStrategy) ||
-          (control.show === 'visible-except-when-other-strategy-is-active' &&
-            (currentStrategy == null || s.id === currentStrategy)),
+    let applicableControls: Array<ControlWithKey> = []
+    let isResizable: boolean = false
+    // Add the controls for currently applicable strategies.
+    for (const strategy of applicableStrategies) {
+      if (isResizableStrategy(strategy)) {
+        isResizable = true
+      }
+      const strategyControls = getApplicableControls(currentStrategy, strategy)
+      applicableControls = addAllUniquelyBy(
+        applicableControls,
+        strategyControls,
+        (l, r) => l.control === r.control,
       )
-      return addAllUniquelyBy(working, filteredControls, (l, r) => l.control === r.control)
-    }, [])
+    }
+    // Special case controls.
+    if (!isResizable) {
+      applicableControls.push(notResizableControls)
+    }
```

Comment:
Hmmm this means that these controls show up during all dragging interactions, which also means they render whilst you mouse down on a selected resizeable element. I think the approach here needs to be that we explicitly choose when to include these controls, rather than implicitly adding them

Reply:
Hmmm, showing up during drags is definitely unexpected. It's a pain because of it being the inverse of an actual condition, where other things are better at determining that case.

Reply:
This should now be fixed in 7ea4e4003.